### DISCUSSION TITLE
Section 3: Unable to render pdf using pandoc & LaTeX, shows the following error everytime

### DISCUSSION THREAD

**Discussion by 0xtdey** [Cyfrin/security-and-auditing-full-course-s23#68] [Q&A]
```javascript
! Emergency stop.
<*> /tmp/tex2pdf.-f76967ed4654acf3/input.tex
                                             
Here is how much of TeX's memory you used:
 16 strings out of 480247
 507 string characters out of 5896152
 283961 words of memory out of 5000000
 18060 multiletter control sequences out of 15000+600000
 469259 words of font info for 28 fonts, out of 8000000 for 9000
 14 hyphenation exceptions out of 8191
 13i,0n,12p,72b,10s stack positions out of 5000i,500n,10000p,200000b,80000s

!  ==> Fatal error occurred, no output PDF file produced!
```

**Comment by johnsonstephan** [comment]
Hi @tamoghna-dey, it appears you may be running into a memory issue. To solve, you can increase the amount of swap space.

- Assuming you are running on a Unix-based system, you can follow along with this [5 minute video](https://www.youtube.com/watch?v=cYhC93hqiEQ) to solve
- Or you can skim [this article](https://www.addictivetips.com/ubuntu-linux-tips/ubuntu-increase-swap-guide/) to quickly find all the commands needed

After you give this a try, let me know if it helps to solve your problem!

**Reply by 0xtdey** [reply]
I tried following the article, and it did show that the space is increased to 8gb but when i try to run the command for generating pdf again, it still shows the same error

**Reply by cromewar** [reply]
@tamoghna-dey did you find any way to solve this?

**Reply by 0xtdey** [reply]
Not yet, swapping the space didn't worked out and it still shows the same error

**Comment by sharonphiliplima** [comment]
![image](https://github.com/Cyfrin/security-and-auditing-full-course-s23/assets/69852010/49e03258-cad1-45d5-b705-853ab3257b30)
Guys, how long should I wait for this [Latex](https://www.tug.org/texlive/quickinstall.html) to download? 
Seems like it can take forever

I followed these
```
Non-interactive default installation on anything but Windows:

cd /tmp # working directory of your choice
Download: wget https://mirror.ctan.org/systems/texlive/tlnet/install-tl-unx.tar.gz
           or: curl -L -o install-tl-unx.tar.gz https://mirror.ctan.org/systems/texlive/tlnet/install-tl-unx.tar.gz
(or via whatever other method you like)
zcat < install-tl-unx.tar.gz | tar xf -
cd install-tl-*
perl ./install-tl --no-interaction # as root or with [writable destination](https://www.tug.org/texlive/quickinstall.html#running)
Finally, prepend /usr/local/texlive/YYYY/bin/PLATFORM to your PATH,
e.g., /usr/local/texlive/2024/bin/x86_64-linux
```

**Reply by EngrPips** [reply]
Hey @tamoghna-dey, It took me around 5 hours to download the Latex gigantic file successfully because my internet connection was slow. So it can take a long time if your internet connection isn't fast.
